src/Faithlife.Build/DotNetBuild.cs

```diff
@@ -336,21 +336,36 @@ IReadOnlyList<string> BuildNuGetPackages()
 
 							var docsPath = Path.Combine(docsRepoDirectory, docsSettings.TargetDirectory ?? "docs");
 
-							string? xmlDocGenPath = null;
-							var xmlDocGenProject = FindFiles("tools/XmlDocGen/XmlDocGen.csproj").FirstOrDefault();
-							if (xmlDocGenProject is not null)
+							var xmlDocGenPaths = new List<string>();
+							var xmlDocGenProjectPath = FindFiles("tools/XmlDocGen/XmlDocGen.csproj").FirstOrDefault();
+							if (xmlDocGenProjectPath is not null)
 							{
-								RunDotNet("publish", xmlDocGenProject,
-									"-c", settings.GetConfiguration(),
-									settings.GetPlatformArg(),
-									"--nologo",
-									"--verbosity", "quiet",
-									"--output", Path.Combine("tools", "bin", "XmlDocGen"));
-								xmlDocGenPath = Path.Combine("tools", "bin", "XmlDocGen", "XmlDocGen.dll");
-								if (!File.Exists(xmlDocGenPath))
-									xmlDocGenPath = Path.Combine("tools", "bin", "XmlDocGen", "XmlDocGen.exe");
-								if (!File.Exists(xmlDocGenPath))
-									throw new BuildException("Failed to build XmlDocGen.");
+								var xmlDocGenProjectDocument = XDocument.Load(xmlDocGenProjectPath);
+								var xmlDocGenFrameworks = GetTargetFrameworks();
+								foreach (var framework in xmlDocGenFrameworks)
+								{
+									RunDotNet("publish", xmlDocGenProjectPath,
+										framework.Length != 0 ? "--framework" : null, framework.Length != 0 ? framework : null,
+										"-c", settings.GetConfiguration(),
+										settings.GetPlatformArg(),
+										"--nologo",
+										"--verbosity", "quiet",
+										"--output", Path.Combine("tools", "bin", framework, "XmlDocGen"));
+									var xmlDocGenPath = Path.Combine("tools", "bin", framework, "XmlDocGen", "XmlDocGen.dll");
+									if (!File.Exists(xmlDocGenPath))
+										xmlDocGenPath = Path.Combine("tools", "bin", framework, "XmlDocGen", "XmlDocGen.exe");
+									if (!File.Exists(xmlDocGenPath))
+										throw new BuildException($"Failed to build XmlDocGen{(framework.Length == 0 ? "" : $" ({framework})")}.");
+									xmlDocGenPaths.Add(xmlDocGenPath);
+								}
+
+								string[] GetTargetFrameworks()
+								{
+									// return single empty framework unless there are more than one
+									var text = xmlDocGenProjectDocument.XPathSelectElements("Project/PropertyGroup/TargetFrameworks").FirstOrDefault()?.Value;
+									var values = text is null ? Array.Empty<string>() : text.Split(';').Select(x => x.Trim()).Where(x => x.Length != 0).ToArray();
```

Comment:
```suggestion
									var values = text is null ? Array.Empty<string>() : text.Split(';', StringSplitOptions.TrimEntries | StringSplitOptions.RemoveEmptyEntries).ToArray();
```

Pity that using the built-in way is even more code...

Reply:
That's not supported in .NET Standard 2.x, unfortunately.